torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -622,9 +622,16 @@ RegisterOperators reg(
              }
 
              // Check that type of the Tensor matches that of the annotation.
-             TORCH_CHECK(
-                 tryScalarTypeFromJitType(out_ty) == t.scalar_type(),
-                 "Output annotation element type and runtime tensor element type must match for tolist()")
+             // Make an exception for the case in which the annotated type is
+             // float and the Tensor data type is also float; the elements will
+             // be casted to double later.
+             auto scalarTypeForJitType = tryScalarTypeFromJitType(out_ty);
+             if (scalarTypeForJitType != at::ScalarType::Double ||
```

Comment:
Would this be better expressed as `!(scalarTypeForJitType == at::ScalarType::Double && t.scalar_type() == at::ScalarType::Float`.